src/bin/pack.rs

```diff
@@ -0,0 +1,362 @@
+                                name = child_name.to_string();
+                            }
+                        }
+                        Ok(_) | Err(_) => {
+                            // use current name because either --no-xattr is set,
+                            // xattr is None, or getting xattr on file (like ._ files) errors
+                            name = child_name.to_string();
+                        }
+                    }
+                    self.depth += 1;
+                    let value = self.pack(child.clone(), &config)?;
+                    self.depth -= 1;
+                    if let Some(value) = value {
+                        entries.insert(name, value);
+                    }
+                }
+
+                Ok(Some(V::from_named_dir(entries, &config)))
+            }
+            "list" => {
+                let mut children = fs::read_dir(path.clone())?
+                    .map(|res| res.map(|e| e.path()))
+                    .collect::<Result<Vec<_>, Error>>()?;
+                children.sort_unstable_by(|a, b| {
+                    a.file_name()
+                        .unwrap()
+                        .to_str()
+                        .unwrap()
+                        .cmp(&b.file_name().unwrap().to_str().unwrap())
+                });
+
+                let mut entries = Vec::with_capacity(children.len());
+
+                for child in children {
+                    let child_name = child.file_name().unwrap().to_str().unwrap();
+                    if config.ignored_file(child_name) {
+                        warn!("skipping ignored file {:?}", child_name);
+                        continue;
+                    }
+                    self.depth += 1;
+                    let value = self.pack(child, &config)?;
+                    self.depth -= 1;
+                    if let Some(value) = value {
+                        entries.push(value);
+                    }
+                }
+
+                Ok(Some(V::from_list_dir(entries, &config)))
+            }
+            typ => {
+                if let Ok(t) = Typ::from_str(typ) {
+                    let file = fs::File::open(&path).unwrap();
+                    let mut reader = BufReader::new(&file);
+                    let mut contents: Vec<u8> = Vec::new();
+                    reader.read_to_end(&mut contents).unwrap();
+                    match String::from_utf8(contents.clone()) {
+                        Ok(mut contents) if t != Typ::Bytes => {
+                            if config.add_newlines && contents.ends_with('\n') {
+                                contents.truncate(contents.len() - 1);
+                            }
+                            Ok(Some(V::from_string(t, contents, &config)))
+                        }
+                        Ok(_) | Err(_) => Ok(Some(V::from_bytes(contents, &config))),
+                    }
+                } else {
+                    error!(
```

Comment:
When we get a type we don't understand, we should revert to `auto`. The `Typ::from_str()` call could come earlier and then we'd have an exhausive match here and a warning up there.